tools/loader/ImageClassifier.cpp

```diff
@@ -301,15 +359,10 @@ int main(int argc, char **argv) {
     loader.runInference(ctx);
 
     // Print out the inferred image classification.
-    auto H = SMVarT->getHandle<>();
-    llvm::outs() << "Model: " << loader.getFunction()->getName() << "\n";
-    for (unsigned i = 0; i < inputImageFilenames.size(); i++) {
-      Tensor slice = H.extractSlice(i);
-      auto SH = slice.getHandle<>();
-      llvm::outs() << " File: " << inputImageFilenames[i];
-
-      auto topKPairs = getTopKPairs(SH);
-      printTopKPairs(topKPairs);
+    if (runUsingFP16) {
+      printResults<float16>(loader, SMVarT);
+    } else {
+      printResults<float>(loader, SMVarT);
```

Comment:
Alternatively you could convert back the tensor to fp32 so you don't have to touch this part.